doc/CI_Hardware.md

```diff
@@ -0,0 +1,243 @@
+As noted in the previous section we have a test.config.toml file that contains information of the test to run for each board in the Tock/Boards directory , which contains each specific Tock supported board as well. Thus to Add tests, there are two primary steps:
+- **Adding Tests**
+    - **1** To add tests, you’ll need to create two files: a main.c (c) file and a test.py (python) file.
+        - The c file will be titled “main” which is the application to be used to test on the target board. For more information on this file and what it does, look at the section How instances work - How tests work.
+        - The python file will be named “test” which is the file that reads into any information from the application, and outputs any messages that are meaningful. This information can be the state, action, or messages from the application onto the target board.
+        - Then, you can create a readme and makefile corresponding to the test to explain what functions are being tested on the board and to compile/build the application (test) for the board.
+        - To add the test, you’ll push the files of the test with the title of the function being tested to the directory path libtock-c/examples/ci-tests in github. The full path will then be libtock-c/examples/ci-tests/{app} with “{app}” being the title to the test you made as the folder. This is necessary because the “main.py” file uses the path to build, install, and run the test.
+    - **2** To add the tests to the boards you want to test, you’ll need to go to the specific board in Tock/Boards directory ([Official Tock Repository](https://github.com/tock/tock/tree/master/boards/nordic) or [Forked Repository  w/ file](https://github.com/goodoomoodoo/tock/tree/master/boards/nordic/nrf52840dk)), and in the boards directory, you’ll either edit, or create, the test.config.toml file for that board. There, you’ll go to the test object, or create one, and either add it to the “all” specifier for all raspberry Pi’s to run the test, or add the test to specific Raspberry Pi’s by making the specifier for the test object be the identity number of said raspberry pi. To see what’s included currently in the test configuration file for each board, visit the nrf52840dk board [here](https://github.com/goodoomoodoo/tock/blob/master/boards/nordic/nrf52840dk/test.config.toml) to see the contents and format. Also revisit the previous section discussing the test configuration file [here](#configuration-files).
+
+![Diagram for Adding/Choosing Tests](tock/doc/images/testsdiagram.png)
+
+# What to do if Something Goes Wrong
+
+## If a test fails
+Take a look at what type of test failed (in the output on github) and see what is the issue of the build of the board accordingly. The output in github currently prints what test failed to give the user a better understanding of what’s wrong in the installation.
+
+## If there’s boot up issues of the Raspberry Pi
+Go to Troubleshooting section [here.](#troubleshoot)
```

Comment:
```suggestion
Go to the [Troubleshooting section](#troubleshoot).
```

Modern accessibility guidelines say to avoid `here` links, as screen readers do not handle them well.